platforms/software/testing-base/src/main/java/org/gradle/api/internal/tasks/testing/report/generic/MetadataRendererRegistry.java

```diff
@@ -123,16 +151,34 @@ public SimpleHtmlWriter render(Object metadata, SimpleHtmlWriter htmlWriter) thr
         }
     }
 
-    public static final class URIMetadataRenderer implements MetadataRenderer<URI> {
+    public static final class ClickableLinkRenderer implements MetadataRenderer {
         @Override
-        public Class<URI> getMetadataType() {
-            return URI.class;
+        public Set<Class<?>> getMetadataTypes() {
+            return new HashSet<>(Arrays.asList(URI.class, File.class, RegularFile.class));
         }
 
         @Override
         public SimpleHtmlWriter render(Object metadata, SimpleHtmlWriter htmlWriter) throws IOException {
-            return (SimpleHtmlWriter) htmlWriter.startElement("a").attribute("href", metadata.toString())
-                .characters(MetadataRenderer.trimIfNecessary(metadata.toString()))
+            String text;
+            String link;
+            if (metadata instanceof File) {
+                link = TextUtil.normaliseFileSeparators(((File) metadata).toURI().toString());
+                text = ((File) metadata).getName();
+            } else if (metadata instanceof RegularFile) {
+                link = TextUtil.normaliseFileSeparators(((RegularFile) metadata).getAsFile().toURI().toString());
+                text = ((RegularFile) metadata).getAsFile().getName();
+            } else {
+                if (((URI) metadata).getScheme().equals("file")) {
+                    link = TextUtil.normaliseFileSeparators(metadata.toString());
+                    text = new File((URI) metadata).getName();
+                } else {
+                    link = metadata.toString();
+                    text = metadata.toString();
+                }
+            }
```

Comment:
I'm confused by the use of `normaliseFileSeparators` here. All of these are URIs, so they should already have `/`, and any `\` should not be changed.

Reply:
Perhaps this isn't needed and I was just being overzealous after it broke on Windows.  I'll test without it.